llrt_utils/src/encoding.rs

```diff
@@ -2,62 +2,67 @@
 // SPDX-License-Identifier: Apache-2.0
 use hex_simd::AsciiCase;
 
-macro_rules! encoder_enum {
-    (
-        $(#[$attr:meta])*
-        pub enum $enum_name:ident {
-            $($variant:ident),* $(,)?
-        }
-    ) => {
-        $(#[$attr])*
-        pub enum $enum_name {
-            $($variant),*
-        }
-
-        impl $enum_name {
-            #[allow(clippy::should_implement_trait)]
-            pub fn from_str(encoding: &str) -> Result<Self, String> {
-
-                let normalized:String = encoding.chars()
-                    .enumerate()
-                    .map(|(i, c)| {
-                        if i == 0 {
-                            c.to_ascii_uppercase()
-                        } else {
-                            c.to_ascii_lowercase()
-                        }
-                    })
-                    .filter(|&c| c != '-' && c != '_')
-                    .collect();
-
-                match normalized.as_str() {
-                    $(
-                        stringify!($variant) => Ok(Self::$variant),
-                    )*
-                    _ => Err(["The \"", &encoding, "\" encoding is not supported"].concat()),
-                }
-            }
-        }
-    };
+use once_cell::sync::Lazy;
+use std::collections::HashMap;
+
+#[derive(Clone)]
+pub enum Encoder {
+    Hex,
+    Base64,
+    Windows1252,
+    Utf8,
+    Utf16le,
 }
 
-encoder_enum! {
-    pub enum Encoder {
-        Hex,
-        Base64,
-        Utf8,
-        Iso88591,
-        Utf16le,
-    }
-}
+static ENCODING_MAP: Lazy<HashMap<&'static str, Encoder>> = Lazy::new(|| {
+    let mut map = HashMap::new();
```

Comment:
We should create with capacity as we know up front how many items to put in there